### PR TITLE
VZ-10879 Update new Verrazzano controller so that it can be used without the other legacy controller.

### PR DESCRIPTION
**Pull request by pfmackin**
Update new Verrazzano controller so that it can be used without the other legacy controller.

1. implement full install, update with pre and post logic
2. implement full upgrade with pre and post logic
3. implement full uninstall with pre and post logic 
4. update VZ conditions and state as needed
5. remove module logic from legacy controller
6. remove controller sync code
7. either start new controller or the legacy controller, not both
8. move custom resource or component code to a custom package

### CODE REVIEW

File: platform-operator/experimental/controllers/verrazzano/custom/istio.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2023, Oracle and/or its affiliates.
```
Comment (michael-cico) [inline platform-operator/experimental/controllers/verrazzano/custom/istio.go:1]: nit, not sure I like the `custom` package name, but I don't have anything better.  Is this all global pre/post handling ?

Comment (pfmackin) [reply platform-operator/experimental/controllers/verrazzano/custom/istio.go:1]: it is all custom code for components, multicluster.  We can rename after if you have a better name.  Ideally it goes away and pushed to components.

Comment (michael-cico) [reply platform-operator/experimental/controllers/verrazzano/custom/istio.go:1]: nah, I'm good ;-)

**Review by michael-cico — APPROVED** [review]

File: platform-operator/experimental/controllers/verrazzano/module.go [hunk 2]
```diff
@@ -8,19 +8,84 @@ import (
...
+		log.ErrorfThrottled("Error loading module catalog: %v", err)
+		return result.NewResultShortRequeueDelayWithError(err)
+	}
+
+	// Create or Update a Module for each enabled resource
+	for _, comp := range registry.GetComponents() {
+		if !comp.IsEnabled(effectiveCR) {
+			continue
+		}
+		if !comp.ShouldUseModule() {
```
Comment (mbenmclean) [inline platform-operator/experimental/controllers/verrazzano/module.go:37]: This can be removed now I think

**Review by mbenmclean — APPROVED** [review]

File: platform-operator/experimental/controllers/verrazzano/module.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/experimental/controllers/verrazzano/module.go:37]: yes, i will do it post merge

**Pull request merged**